corellium/domain/src/test/kotlin/flank/corellium/domain/run/test/android/step/ExecuteTestsKtTest.kt

```diff
@@ -0,0 +1,116 @@
+package flank.corellium.domain.run.test.android.step
+
+import flank.apk.Apk
+import flank.corellium.api.CorelliumApi
+import flank.corellium.domain.RunTestCorelliumAndroid
+import flank.corellium.domain.RunTestCorelliumAndroid.ExecuteTests
+import flank.corellium.domain.RunTestCorelliumAndroid.ExecuteTests.ADB_LOG
+import flank.corellium.domain.invalidLog
+import flank.corellium.domain.stubCredentials
+import flank.corellium.domain.validLog
+import flank.junit.JUnit
+import flank.log.Event
+import flank.log.Output
+import flank.shard.Shard
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.asFlow
+import kotlinx.coroutines.flow.flattenConcat
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.flow.onStart
+import kotlinx.coroutines.runBlocking
+import org.junit.After
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+import java.io.File
+
+class ExecuteTestsKtTest : RunTestCorelliumAndroid.Context {
+
+    override lateinit var api: CorelliumApi
+    override var out: Output = ::println
+    override val apk = Apk.Api()
+    override val junit = JUnit.Api()
+    override val args = RunTestCorelliumAndroid.Args(
+        credentials = stubCredentials,
+        apks = emptyList(),
+        maxShardsCount = 1,
+    )
+
+    private val dir = File(args.outputDir)
```

Comment:
Instead of `File` use `TemporaryFolder` (JUnit). Current implementation does not guarantee clean up after failed tests/exception thrown

Reply:
> Instead of `File` use `TemporaryFolder` (JUnit). Current implementation does not guarantee clean up after failed tests/exception thrown

I do not agree. This is the first shot from duckduckgo for the phase `JUnit After`:  https://junit.org/junit4/javadoc/4.12/org/junit/After.html

Reply:
And this is the short from google (I do not use duckduckgo, maybe I should start to do so)
https://junit.org/junit4/javadoc/4.12/org/junit/rules/TemporaryFolder.html
 
`@After` may throw an exception itself which will break i.e. recursive deletion. 
Not saying your implementation will not work, most likely it will be working fine.
`TemporaryFolder` is implemented to be used in exactly the same cases we have in this particular test.

Reply:
Ok, so according to documentation, the `TemporaryFolder` is also not guaranteed clean up same as `@After`. Additionally, it will not throw an exception on recursive deletion fail. This is bad because the test should always clean up its own garbages, if it is not possible that means probably some external process is reading the files created during the test. This situation is an edge case that should throw an exception.